cumulusci/tasks/bulkdata.py

```diff
@@ -1,63 +1,119 @@
+class BulkJobTaskMixin(object):
+    def _job_state_from_batches(self, job_id):
+        uri = urljoin(self.bulk.endpoint + "/", "job/{0}/batch".format(job_id))
+        response = requests.get(uri, headers=self.bulk.headers())
+        return self._parse_job_state(response.content)
+
+    def _parse_job_state(self, xml):
+        tree = ET.fromstring(xml)
+        completed = 0
+        pending = 0
+        failed = 0
+        for el in tree.iterfind(".//{%s}state" % self.bulk.jobNS):
+            state = el.text
+            if state == "Not Processed":
+                return "Aborted"
+            elif state == "Failed":
+                failed += 1
+            elif state == "Completed":
+                completed += 1
+            else:  # Queued, InProgress
+                pending += 1
+        if pending:
+            return "InProgress"
+        elif failed:
+            return "Failed"
+        else:
+            return "Completed"
+
+    def _wait_for_job(self, job_id):
+        while True:
+            job_status = self.bulk.job_status(job_id)
+            self.logger.info(
+                "    Waiting for job {} ({}/{})".format(
+                    job_id,
+                    job_status["numberBatchesCompleted"],
+                    job_status["numberBatchesTotal"],
+                )
+            )
+            result = self._job_state_from_batches(job_id)
+            if result != "InProgress":
+                break
+            time.sleep(10)
+        self.logger.info("Job {} finished with result: {}".format(job_id, result))
+        return result
+
+    def _sql_bulk_insert_from_csv(self, conn, table, columns, data_file):
+        if conn.dialect.name == "psycopg2":
+            # psycopg2 (the postgres driver) supports COPY FROM
+            # to efficiently bulk insert rows in CSV format
+            with conn.connection.cursor() as cursor:
+                cursor.copy_expert(
+                    "COPY {} ({}) FROM STDIN WITH (FORMAT CSV)".format(
+                        table, ",".join(columns)
+                    ),
+                    data_file,
+                )
+        else:
+            # For other db drivers we need to use standard SQL
+            # -- this is optimized for ease of implementation
+            # rather than performance and may need more work.
+            reader = unicodecsv.DictReader(data_file, columns)
+            table = self.metadata.tables[table]
+            rows = list(reader)
+            if rows:
+                conn.execute(table.insert().values(rows))
+        self.session.flush()
 
 
-class DeleteData(BaseSalesforceApiTask):
+class DeleteData(BaseSalesforceApiTask, BulkJobTaskMixin):
 
     task_options = {
         "objects": {
             "description": "A list of objects to delete records from in order of deletion.  If passed via command line, use a comma separated string",
             "required": True,
-        }
+        },
+        "operation": {
```

Comment:
I'd rather see this as a soft_delete boolean option that defaults to false. is there a reason to allow a string operation name?